cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -31,7 +31,9 @@
 
 @pytest.fixture(scope="module")
 def vm_name():
-    return "test_snpsht_" + fauxfactory.gen_alphanumeric()
```

Comment:
I do need to change the scope, yes. I did try, it didn't work. What you see here is just a workaround, the underlying problem is a bit different.

There's a PR with new snapshot test #5003 . The test is parametrized and it's using module-scoped test_vm. Either I'm not getting the scopes at all, or it doesn't work, because instead of creating one VM for the test with all it's parametrization, it creates a new VM each time the test is run with different parameter. This means that the VM gets created, the test is run once with one parameter, then the VM is destroyed and immediately created again, the test is run the second time, and so on. Because of this, there were issues with multiple VMs in CFME with the same name, some of which were orphaned, and the automation was choosing the already destroyed one.

This of course isn't enough trouble, while running this I was hitting and issue with virtualcenter.delete_vm() https://github.com/ManageIQ/wrapanapi/issues/154 . Can't reproduce that locally, it does happen with PRT. Can't run PRT, because I need this PR to be merged first.

My solution is what you are looking at now. I turned my back on scopes, changed fixtures so that the VMs have always different name, problem went away.